edge-modules/api-proxy-module/src/monitors/certs_monitor.rs

```diff
@@ -76,6 +77,27 @@ pub fn start(
                     false
                 }
             };
+        }
+
+        //Trust bundle just received. Request for a reset of the API proxy.
+        notify_certs_rotated.notify();
+
+        //Loop to check if server certificate expired.
+        //It is implemented as a polling instead of a delay until certificate expiry, because clocks are unreliable.
+        //If the system clock gets readjusted while the task is sleeping, the system might wake up after the certificate expiry.
+        loop {
+            let wait_shutdown = shutdown_signal.notified();
+            futures::pin_mut!(wait_shutdown);
+
+            match futures::future::select(time::delay_for(CERTIFICATE_POLL_INTERVAL), wait_shutdown)
```

Comment:
nit: it is a bit odd to use such constructs `Either::Left(_) => {}`. It is better to use `if let Either::Right(_) = ...` in that case...